client/src/index.jsx

```diff
@@ -117,18 +149,39 @@ class App extends React.Component {
     let targetValue = e.target.value;
     const currentValues = Object.assign({}, this.state.currentValues);
 
+    if(targetValue === ''){
+      targetValue = 0;
+    }
+
     if (targetName !== 'loan_program') {
       targetValue = parseFloat(targetValue);
     }
 
+    if(targetValue.toString().length > 9){
+      targetValue = 0;
+      alert('invalid input');
+    }
+
+    if(targetName !== 'home_price' &&  targetValue > this.state.currentValues.home_price){
+      targetValue = 0;
+      alert('invalid input');
+    }
+
+    if(targetName === 'home_price'){
```

Comment:
space between parens
```suggestion
    if (targetName === 'home_price') {
```